src/common/checkDomain.f90

```diff
@@ -19,24 +19,27 @@ module checkDomainML
   implicit none
   private
 
-  public checkDomain
+  public :: constrain_to_domain
```

Comment:
I don't think this name is better thant the old one. We constrain in z, but we remove in xy.
Strange to not have same name in filename than in subroutine doing only one thing.

Reply:
I disagree, the naming of the function suggests we check the domain itself, not that we act on the particle in/outside the domain

Reply:
Using the more active name `check_in_domain`